docs/source/user_guide_pipeline.rst

```diff
@@ -0,0 +1,64 @@
+.. _user-guide-pipeline:
+
+User Guide: Pipeline
+####################
+
+This page provides information about how to create a pipeline.
+
+
+******************************
+Creating Components
+******************************
+
+Components are units of work that perform simple tasks, such as chunking documents
+or saving results to Neo4j. This package includes a few default components,
+but developers can create their own by following these steps:
```

Comment:
Should we add a brief note mentioning along the lines of "Please ensure that your environment supports asynchronous operations"?